vshard/util.lua

```diff
@@ -279,6 +279,24 @@ local function fiber_cond_wait(cond, timeout)
     return nil, lerror.make(err)
 end
 
+--
+-- Exception-safe future wait
+--
+local function future_wait(cond, timeout)
+    local res, err
+    -- Handle negative timeouts in order to avoid usage error.
+    if timeout and timeout < 0 then
+        err = lerror.timeout()
+        goto fail
+    end
+    res, err = cond:wait_result(timeout)
+    if not err then
+        return res
+    end
```

Comment:
As I said in commit message, errors are inconsistent. Do we want them to be?

Frankly speaking, I don't see it as a big problem, as tnt is a one big inconsistency. If we ever need to deal with timeout error processing, we'll anyway have to use the function, similar to `vtest.error_is_timeout`.

But if we want it, here's the way we may do it:

```diff
diff --git a/vshard/util.lua b/vshard/util.lua
index 45abca2..6e212b2 100644
--- a/vshard/util.lua
+++ b/vshard/util.lua
@@ -293,6 +293,10 @@ local function future_wait(cond, timeout)
     if not err then
         return res
     end
+
+    if err.type == 'TimedOut' then
+        err = lerror.timeout()
+    end
 ::fail::
     return nil, err
 end

```